### PR TITLE
Add NTLM mediator tool to Integration studio

### PR DESCRIPTION
**Pull request by imsuneth**

### CODE REVIEW

File: components/esb-tools/plugins/org.wso2.integrationstudio.gmf.esb.diagram/resources/schema/mediators/other/ntlm.xsd [hunk 1]
```diff
@@ -0,0 +1,37 @@
+<?xml version="1.0" encoding="ISO-8859-1"?>
+<!--
+ Copyright (c) 2018, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
```
Comment (sajithaliyanage) [inline components/esb-tools/plugins/org.wso2.integrationstudio.gmf.esb.diagram/resources/schema/mediators/other/ntlm.xsd:3]: Shall we change the year as 2021?

File: components/esb-tools/plugins/org.wso2.integrationstudio.gmf.esb.diagram/src/org/wso2/integrationstudio/gmf/esb/diagram/custom/deserializer/NTLMMediatorDeserializer.java [hunk 2]
```diff
@@ -0,0 +1,56 @@
+package org.wso2.integrationstudio.gmf.esb.diagram.custom.deserializer;
```
Comment (sajithaliyanage) [inline components/esb-tools/plugins/org.wso2.integrationstudio.gmf.esb.diagram/src/org/wso2/integrationstudio/gmf/esb/diagram/custom/deserializer/NTLMMediatorDeserializer.java:1]: Let's add the WSO2 license header for all the manually created files. Ex: .java, .xsd, etc.

File: components/esb-tools/plugins/org.wso2.integrationstudio.gmf.esb.diagram/resources/schema/mediators/other/ntlm.xsd (hunk 1 above)
Comment (imsuneth) [reply components/esb-tools/plugins/org.wso2.integrationstudio.gmf.esb.diagram/resources/schema/mediators/other/ntlm.xsd]: Sure

File: components/esb-tools/plugins/org.wso2.integrationstudio.gmf.esb.diagram/src/org/wso2/integrationstudio/gmf/esb/diagram/custom/deserializer/NTLMMediatorDeserializer.java (hunk 2 above)
Comment (imsuneth) [reply components/esb-tools/plugins/org.wso2.integrationstudio.gmf.esb.diagram/src/org/wso2/integrationstudio/gmf/esb/diagram/custom/deserializer/NTLMMediatorDeserializer.java:19]: Added

**Review by rosensilva — APPROVED** [review]

File: components/esb-tools/plugins/org.wso2.integrationstudio.gmf.esb.diagram/src/org/wso2/integrationstudio/gmf/esb/diagram/providers/assistants/EsbModelingAssistantProviderOfNTLMMediatorEditPart.java [hunk 3]
```diff
@@ -0,0 +1,28 @@
+package org.wso2.integrationstudio.gmf.esb.diagram.providers.assistants;
```
Comment (chanikag) [inline components/esb-tools/plugins/org.wso2.integrationstudio.gmf.esb.diagram/src/org/wso2/integrationstudio/gmf/esb/diagram/providers/assistants/EsbModelingAssistantProviderOfNTLMMediatorEditPart.java:1]: Let's add license header

File: components/esb-tools/plugins/org.wso2.integrationstudio.gmf.esb.diagram/src/org/wso2/integrationstudio/gmf/esb/diagram/providers/assistants/EsbModelingAssistantProviderOfNTLMMediatorInputConnectorEditPart.java [hunk 4]
```diff
@@ -0,0 +1,169 @@
+package org.wso2.integrationstudio.gmf.esb.diagram.providers.assistants;
```
Comment (chanikag) [inline components/esb-tools/plugins/org.wso2.integrationstudio.gmf.esb.diagram/src/org/wso2/integrationstudio/gmf/esb/diagram/providers/assistants/EsbModelingAssistantProviderOfNTLMMediatorInputConnectorEditPart.java:1]: Let's add license header

File: components/esb-tools/plugins/org.wso2.integrationstudio.gmf.esb.diagram/src/org/wso2/integrationstudio/gmf/esb/diagram/providers/assistants/EsbModelingAssistantProviderOfNTLMMediatorOutputConnectorEditPart.java [hunk 5]
```diff
@@ -0,0 +1,514 @@
+package org.wso2.integrationstudio.gmf.esb.diagram.providers.assistants;
```
Comment (chanikag) [inline components/esb-tools/plugins/org.wso2.integrationstudio.gmf.esb.diagram/src/org/wso2/integrationstudio/gmf/esb/diagram/providers/assistants/EsbModelingAssistantProviderOfNTLMMediatorOutputConnectorEditPart.java:1]: Let's add license header

File: components/esb-tools/plugins/org.wso2.integrationstudio.gmf.esb.persistence/src/org/wso2/integrationstudio/gmf/esb/internal/persistence/NTLMMediatorTransformer.java [hunk 6]
```diff
@@ -0,0 +1,106 @@
...
+            ntlmMediator.setNtlmVersion("");
+        }
+        
+        return ntlmMediator;
+    }
+
+    @Override
+    public void createSynapseObject(TransformationInfo info, EObject subject, List<Endpoint> endPoints)
+            throws TransformerException {
+        // TODO Auto-generated method stub
```
Comment (chanikag) [inline components/esb-tools/plugins/org.wso2.integrationstudio.gmf.esb.persistence/src/org/wso2/integrationstudio/gmf/esb/internal/persistence/NTLMMediatorTransformer.java:90]: Shall we remove this todo?

File: components/esb-tools/plugins/org.wso2.integrationstudio.gmf.esb.diagram/src/org/wso2/integrationstudio/gmf/esb/diagram/providers/assistants/EsbModelingAssistantProviderOfNTLMMediatorEditPart.java (hunk 3 above)
Comment (imsuneth) [reply components/esb-tools/plugins/org.wso2.integrationstudio.gmf.esb.diagram/src/org/wso2/integrationstudio/gmf/esb/diagram/providers/assistants/EsbModelingAssistantProviderOfNTLMMediatorEditPart.java:1]: Fixed https://github.com/wso2/integration-studio/issues/1106

File: components/esb-tools/plugins/org.wso2.integrationstudio.gmf.esb.diagram/src/org/wso2/integrationstudio/gmf/esb/diagram/providers/assistants/EsbModelingAssistantProviderOfNTLMMediatorInputConnectorEditPart.java (hunk 4 above)
Comment (imsuneth) [reply components/esb-tools/plugins/org.wso2.integrationstudio.gmf.esb.diagram/src/org/wso2/integrationstudio/gmf/esb/diagram/providers/assistants/EsbModelingAssistantProviderOfNTLMMediatorInputConnectorEditPart.java:1]: Fixed https://github.com/wso2/integration-studio/issues/1106

File: components/esb-tools/plugins/org.wso2.integrationstudio.gmf.esb.persistence/src/org/wso2/integrationstudio/gmf/esb/internal/persistence/NTLMMediatorTransformer.java (hunk 6 above)
Comment (imsuneth) [reply components/esb-tools/plugins/org.wso2.integrationstudio.gmf.esb.persistence/src/org/wso2/integrationstudio/gmf/esb/internal/persistence/NTLMMediatorTransformer.java:90]: Fixed https://github.com/wso2/integration-studio/issues/1106

**Pull request merged**